File: Client/ruby/get_data/track/track_request_builder.rb
require 'microsoft_kiota_abstractions'
require_relative '../../api_sdk'
require_relative '../get_data'
require_relative './track'

module ApiSdk
    module GetData
        module Track
            ## 
            # Builds and executes requests for operations under \getData\track
            class TrackRequestBuilder < MicrosoftKiotaAbstractions::BaseRequestBuilder
                
                ## 
                ## Instantiates a new TrackRequestBuilder and sets the default values.
                ## @param path_parameters Path parameters for the request
                ## @param request_adapter The request adapter to use to execute the requests.
                ## @return a void
                ## 
                def initialize(path_parameters, request_adapter)
                    super(path_parameters, request_adapter, "{+baseurl}/getData/track?genome={genome}&track={track}{&chrom*,end*,hubUrl*,jsonOutputArrays*,maxItemsOutput*,start*}")
                end
                ## 
                ## @param request_configuration Configuration for the request such as headers, query parameters, and middleware options.
                ## @return a Fiber of binary
                ## 
                def get(request_configuration=nil)
                    request_info = self.to_get_request_information(
                        request_configuration
                    )
                    return @request_adapter.send_async(request_info, Binary, nil)
                end
                ## 
                ## @param request_configuration Configuration for the request such as headers, query parameters, and middleware options.
                ## @return a request_information
                ## 
                def to_get_request_information(request_configuration=nil)
                    request_info = MicrosoftKiotaAbstractions::RequestInformation.new()
                    unless request_configuration.nil?
                        request_info.add_headers_from_raw_object(request_configuration.headers)
                        request_info.set_query_string_parameters_from_raw_object(request_configuration.query_parameters)
                        request_info.add_request_options(request_configuration.options)
                    end
                    request_info.url_template = @url_template
                    request_info.path_parameters = @path_parameters
                    request_info.http_method = :GET
                    return request_info
                end
                ## 
                ## Returns a request builder with the provided arbitrary URL. Using this method means any other path or query parameters are ignored.
                ## @param raw_url The raw URL to use for the request builder.
                ## @return a track_request_builder
                ## 
                def with_url(raw_url)
                    raise StandardError, 'raw_url cannot be null' if raw_url.nil?
                    return TrackRequestBuilder.new(raw_url, @request_adapter)
                end

                class TrackRequestBuilderGetQueryParameters
                    
                    attr_accessor :chrom
                    attr_accessor :end_escaped
                    attr_accessor :genome
                    attr_accessor :hub_url
                    attr_accessor :json_output_arrays
                    attr_accessor :max_items_output
                    attr_accessor :start
                    attr_accessor :track
                    ## 
                    ## Maps the query parameters names to their encoded names for the URI template parsing.
                    ## @param original_name The original query parameter name in the class.
                    ## @return a string
                    ## 
                    def get_query_parameter(original_name)
                        raise StandardError, 'original_name cannot be null' if original_name.nil?
                        case original_name
                            when "hub_url"
                                return "hubUrl"
                            when "json_output_arrays"
                                return "jsonOutputArrays"
                            when "max_items_output"
                                return "maxItemsOutput"
                            else
                                return original_name
                        end
                    end
                end
            end
        end
    end
end

File: Client/java/apisdk/getdata/track/TrackRequestBuilder.java
package apisdk.getdata.track;

import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.BaseRequestConfiguration;
import com.microsoft.kiota.HttpMethod;
import com.microsoft.kiota.QueryParameters;
import com.microsoft.kiota.RequestAdapter;
import com.microsoft.kiota.RequestInformation;
import com.microsoft.kiota.RequestOption;
import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParsableFactory;
import java.io.InputStream;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /getData/track
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class TrackRequestBuilder extends BaseRequestBuilder {
    /**
     * Instantiates a new {@link TrackRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public TrackRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/getData/track?genome={genome}&track={track}{&chrom*,end*,hubUrl*,jsonOutputArrays*,maxItemsOutput*,start*}", pathParameters);
    }
    /**
     * Instantiates a new {@link TrackRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public TrackRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/getData/track?genome={genome}&track={track}{&chrom*,end*,hubUrl*,jsonOutputArrays*,maxItemsOutput*,start*}", rawUrl);
    }
    /**
     * @return a {@link InputStream}
     */
    @jakarta.annotation.Nullable
    public InputStream get() {
        return get(null);
    }
    /**
     * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
     * @return a {@link InputStream}
     */
    @jakarta.annotation.Nullable
    public InputStream get(@jakarta.annotation.Nullable final java.util.function.Consumer<GetRequestConfiguration> requestConfiguration) {
        final RequestInformation requestInfo = toGetRequestInformation(requestConfiguration);
        return this.requestAdapter.sendPrimitive(requestInfo, null, InputStream.class);
    }
    /**
     * @return a {@link RequestInformation}
     */
    @jakarta.annotation.Nonnull
    public RequestInformation toGetRequestInformation() {
        return toGetRequestInformation(null);
    }
    /**
     * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
     * @return a {@link RequestInformation}
     */
    @jakarta.annotation.Nonnull
    public RequestInformation toGetRequestInformation(@jakarta.annotation.Nullable final java.util.function.Consumer<GetRequestConfiguration> requestConfiguration) {
        final RequestInformation requestInfo = new RequestInformation(HttpMethod.GET, urlTemplate, pathParameters);
        requestInfo.configure(requestConfiguration, GetRequestConfiguration::new, x -> x.queryParameters);
        return requestInfo;
    }
    /**
     * Returns a request builder with the provided arbitrary URL. Using this method means any other path or query parameters are ignored.
     * @param rawUrl The raw URL to use for the request builder.
     * @return a {@link TrackRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public TrackRequestBuilder withUrl(@jakarta.annotation.Nonnull final String rawUrl) {
        Objects.requireNonNull(rawUrl);
        return new TrackRequestBuilder(rawUrl, requestAdapter);
    }
    @jakarta.annotation.Generated("com.microsoft.kiota")
    public class GetQueryParameters implements QueryParameters {
        @jakarta.annotation.Nullable
        public String chrom;
        @jakarta.annotation.Nullable
        public Long end;
        @jakarta.annotation.Nullable
        public String genome;
        @jakarta.annotation.Nullable
        public String hubUrl;
        @jakarta.annotation.Nullable
        public Boolean jsonOutputArrays;
        @jakarta.annotation.Nullable
        public Integer maxItemsOutput;
        @jakarta.annotation.Nullable
        public Long start;
        @jakarta.annotation.Nullable
        public String track;
        /**
         * Extracts the query parameters into a map for the URI template parsing.
         * @return a {@link Map<String, Object>}
         */
        @jakarta.annotation.Nonnull
        public Map<String, Object> toQueryParameters() {
            final Map<String, Object> allQueryParams = new HashMap();
            allQueryParams.put("chrom", chrom);
            allQueryParams.put("end", end);
            allQueryParams.put("genome", genome);
            allQueryParams.put("hubUrl", hubUrl);
            allQueryParams.put("jsonOutputArrays", jsonOutputArrays);
            allQueryParams.put("maxItemsOutput", maxItemsOutput);
            allQueryParams.put("start", start);
            allQueryParams.put("track", track);
            return allQueryParams;
        }
    }
    /**
     * Configuration for the request such as headers, query parameters, and middleware options.
     */
    @jakarta.annotation.Generated("com.microsoft.kiota")
    public class GetRequestConfiguration extends BaseRequestConfiguration {
        /**
         * Request query parameters
         */
        @jakarta.annotation.Nullable
        public GetQueryParameters queryParameters = new GetQueryParameters();
    }
}
